.github/actions/nf-test-shard/action.yml

```diff
@@ -0,0 +1,60 @@
+name: "Get number of shards"
+description: "Get the number of nf-test shards for the current CI job"
+inputs:
+  tags:
+    description: "Tags to pass as argument for nf-test --tag parameter"
+    required: true
+  max_shards:
+    description: "Maximum number of shards allowed"
+    required: true
+outputs:
+  shard:
+    description: "Array of shard numbers"
+    value: ${{ steps.shards.outputs.shard }}
+  total_shards:
+    description: "Total number of shards"
+    value: ${{ steps.shards.outputs.total_shards }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install nf-test
+      uses: nf-core/setup-nf-test@v1
+      with:
+        version: ${{ env.NFT_VER }}
+        install-pdiff: true
+
+    - name: Get number of shards
+      id: shards
+      shell: bash
+      run: |
+        # Run nf-test to get the number of related tests
+        nftest_output=$(nf-test test --dry-run --changed-since HEAD^ --filter pipeline --tag ${{ inputs.tags }})
+        echo "nf-test dry-run output: $nftest_output"
+
+        # Default values for shard and total_shards
+        shard="[]"
+        total_shards=0
+
+        # Check if there are related tests
+        if echo "$nftest_output" | grep -q 'Nothing to do'; then
+          echo "No related tests found."
+        else
+          # Extract the number of related tests
+          number_of_shards=$(echo "$nftest_output" | sed -n 's|.*Executed \([0-9]*\) tests.*|\1|p')
```

Comment:
Here we use `shards  = tests` strategy. This is good for long running pipeline level tests. For modules, we might want to switch to `shards  = tests / n ` as the runtime of a single module is generally comparable to action setup time.